Intro overhaul (#42)
oetiker/lshort: book/src/deprecated.tex

% !TEX root = ./lshort.tex

> \chapter{Things You Shouldn't Use}

Reviewer: ```suggestion
\chapter{Things You Shouldn't Use}

\begin{intro}
  \LaTeX{} has been around for well over 30 years now. And it has remained largly backward compatible. This means
  you can scrib use text and tricks from a friends document written ten years ago and use them in your document today.
  In many intances this is totally fine, in others, while \LaTeX{} works fine with the old code, you still should not do it. 
  Have a look!
\end{intro}

```

  \vchto|\newcommand{\foo}[4][bar]{ ... }|
  |\NewDocumentCommand{\foo}{O{bar}mmmm}{ ... }|
\end{chktexignore}

The \ci{def} command is a \hologo{plainTeX} primitive. It \emph{always} defines
the command, even if it was already defined. This is usually not desirable, but
if it's needed you can use \ci{DeclareDocumentCommand}.
\begin{chktexignore}
  \vchto|\def\foo#1#2#3{ ... }|
  |\NewDocumentCommand{\foo}{mmm}{ ... }|
\end{chktexignore}

\section{Copying Commands}
\instead{
  \ci{let}
} {
  \ci{NewCommandCopy} \\
  \ci{RenewCommandCopy} \\
  \ci{DeclareCommandCopy}
}

Similarly to~\ref{sec:def}, \ci{let} is a \hologo{plainTeX} primitive that does
does not guard against accidental redefinition. Moreover it does not work
correctly with some \hologo{LaTeX} commands. Use \ci{NewCommandCopy} as
described in Section~\ref{sec:copyingcommands}.
\chto|\let\foo\bar||\NewCommandCopy\foo\bar|

\section{Aligning Equations}
\instead{
  \ei*{eqnarray} \\
  \ei*{eqnarray*}
}{
  \ei*{align}  \\
  \ei*{align*} \\
  (Both in \pai{amsmath}.)
}

\ei{eqnarray} and \ei{eqnarray*} are \hologo{LaTeX} environments that allow
aligning equations. However spacing around the binary operators in these is far

> from ideal. Therefore it is recommended to always use \ei{align} environments

Reviewer: ```suggestion
from ideal. Therefore it is recommended to always use the \ei{align} environments
```

% !TEX root = ./lshort.tex
\chapter{Things You Shouldn't Use}

\begin{intro}
  \LaTeX{} has been around for well over 30 years now. And it has remained

>   largely backward compatible. This means you can use text and tricks from

Reviewer: ```suggestion
  largely backward compatible. This means you can easily steal snipptes from
```

  a friends document written ten years ago and use them in your document today.

>   In many instances this is totally fine, in others, while \LaTeX{} works fine

Reviewer: ```suggestion
  In many instances this work without any problem. But eventhough \LaTeX{} seems happy
```

}

The \TeX{} version of entering quotes was to rely on ligatures for the given
quotation mark. This method is not context aware and cannot be customised.
Using \pai{csquotes} package as described in Section~\ref{sec:csquotes} allows
greater control over the typesetting of the quotations.

\chto|``quote''||\usepackage{csquotes}
% ...
\enquote{quote}|

\section{Printing Verbatim}
\instead{
\ei{verbatim}
}{

> \ei{verbatim} (from \pai(verbatim) package)

Reviewer: ```suggestion
\ei{verbatim} (from the \pai(verbatim) package)
```

}

> \LaTeX{} defines its own \ei{verbatim} environment, however there are problems

Reviewer: ```suggestion
\LaTeX{} comes with a \ei{verbatim} environment built into the core, however there are problems
```

with it. Firstly due to the way delimiting is handled you cannot put spaces
\begin{chktexignore}
  between \mintinline{latex}|\end| and \mintinline{latex}|{verbatim}|.
\end{chktexignore}
Second and more important problem is that it reads the whole input at once
which risks overflowing \TeX{}'s memory. The \pai{verbatim} package described
in Section~\ref{sec:verbatim} solves both of these problems.
\begin{chktexignore}
\chto|\begin{verbatim}
  code
\end{verbatim}||\usepackage{verbatim}
% ...
\begin{verbatim}
  code
\end{verbatim}|
\end{chktexignore}

> \section{Adding Bibliography}

Reviewer: ```suggestion
\section{Adding a Bibliography}
```
